reflex/utils/prerequisites.py

```diff
@@ -276,6 +267,47 @@ def remove_existing_bun_installation():
         path_ops.rm(os.path.expandvars(constants.BUN_ROOT_PATH))
 
 
+def validate_and_install_node():
+    """Validate nodejs have install or not."""
+    if not check_node_version():
+        install_node()
+
+
+def install_node():
+    """Install nvm and nodejs onto the user's system.
+
+
+    Raises:
+        FileNotFoundError: if unzip or curl packages are not found.
+        Exit: if installation failed
+    """
+    if platform.system() == "Linux":
+        # Only install if bun is not already installed.
+        console.log("Installing nvm...")
+
+        # Check if curl is installed
+        curl_path = path_ops.which("curl")
+        if curl_path is None:
+            raise FileNotFoundError("Reflex requires curl to be installed.")
+
+        result = subprocess.run(constants.INSTALL_NVM, shell=True)
+
+        if result.returncode != 0:
+            raise typer.Exit(code=result.returncode)
+
+        console.log("Installing node...")
+        result = subprocess.run(constants.INSTALL_NODE, shell=True)
```

Comment:
After installing node, it would be nice to reload the shell configuration files (`.zshrc`, `.bashrc`) as the node command will not be immediately available until the shell is restarted. This means reflex will always try to install nvm and node on `reflex init` until shell is restarted

Reply:
In a follow up we will move everything to a `~/.reflex` folder so the user won't have to manage their own node or bun dependencies.